tests/listobjects/listobjects.go

```diff
@@ -115,23 +119,62 @@ func runTests(t *testing.T, schemaVersion string, client ListObjectsClientInterf
 					if assertion.ErrorCode == 0 {
 						require.NoError(t, err)
 						require.Subset(t, assertion.Expectation, resp.Objects)
+					} else {
+						require.Error(t, err)
+						e, ok := status.FromError(err)
+						require.True(t, ok)
+						require.Equal(t, assertion.ErrorCode, int(e.Code()))
+					}
+
+					// assert 2: on streaming list objects endpoint
+					done := make(chan struct{})
+					var streamedObjectIds []string
 
-						for _, object := range resp.Objects {
-							// each object in the response of ListObjects should return check -> true
-							checkResp, err := client.Check(ctx, &pb.CheckRequest{
-								StoreId:          storeID,
-								TupleKey:         tuple.NewTupleKey(object, assertion.Request.Relation, assertion.Request.User),
-								ContextualTuples: assertion.Request.ContextualTuples,
-							})
-							require.NoError(t, err)
-							require.True(t, checkResp.Allowed, fmt.Sprintf("Expected Check(%s#%s@%s) to be true, got false", object, assertion.Request.Relation, assertion.Request.User))
+					clientStream, err := client.StreamedListObjects(ctx, &pb.StreamedListObjectsRequest{
+						StoreId:          storeID,
+						Type:             assertion.Request.Type,
+						Relation:         assertion.Request.Relation,
+						User:             assertion.Request.User,
+						ContextualTuples: assertion.Request.ContextualTuples,
+					}, []grpc.CallOption{}...)
+					go func() {
+						for {
+							resp, err := clientStream.Recv()
+							if err != nil {
+								if !errors.Is(err, io.EOF) {
+									require.Failf(t, "Streaming ListObjects threw an unexpected error", err.Error())
+								}
+								break
+							}
+							streamedObjectIds = append(streamedObjectIds, resp.Object)
 						}
+						done <- struct{}{}
+					}()
+					<-done
+
+					if assertion.ErrorCode == 0 {
+						require.NoError(t, err)
+						require.ElementsMatch(t, assertion.Expectation, streamedObjectIds)
 					} else {
 						require.Error(t, err)
 						e, ok := status.FromError(err)
 						require.True(t, ok)
 						require.Equal(t, assertion.ErrorCode, int(e.Code()))
 					}
+
+					if assertion.ErrorCode != 0 {
+						continue
+					}
+					// assert 3: each object in the response of ListObjects should return check -> true
+					for _, object := range resp.Objects {
```

Comment:
Instead of `if assertion.ErrorCode != 0 { continue }` above, do you mind putting this for loop in the if statement above?

```
if assertion.ErrorCode == 0 {
    ...
    
    for _, object := range resp.Objects {
        checkResp, err := client.Check(...)
        require.NoError(t, err)
        require.True(t, ...)
    }
}
```